### PR TITLE
Add vagrant-libvirt support

### PR DESCRIPTION
**Pull request by jsafrane**
I use libvirt instead of VirtualBox and these patches helped me to run kube-up.sh on Fedora 21. Detailed steps are in the third patch.

### CODE REVIEW

**Comment by jsafrane** [comment]
> It looks like this may be your first contribution to a Google open source project

It's my second contribution, see #6350.

> If you signed the CLA as a corporation, please let us know the company's name.

I work for Red Hat, Inc.

File: docs/getting-started-guides/vagrant.md [hunk 1]
```diff
@@ -66,6 +76,24 @@ vagrant ssh minion-1
 [vagrant@kubernetes-minion-1] $ sudo journalctl -r -u kubelet
 ```
 
+For some vagrant operations, environment variables form config-default.sh must be set up:
```
Comment (vishh) [inline docs/getting-started-guides/vagrant.md]: s/form/from?

File: docs/getting-started-guides/vagrant.md [hunk 2]
```diff
@@ -66,6 +76,24 @@ vagrant ssh minion-1
...
+```
+export NUM_MINIONS=3
+source cluster/vagrant/config-default.sh
+vagrant status
+```
+
+### Customization
+Following environment variables can be used to tun up your vagrant setup.
```
Comment (vishh) [inline docs/getting-started-guides/vagrant.md]: s/tun/turn?

**Comment by vishh** [comment]
@jsafrane: Will you be able to maintain support for "vagrant-libvirt" in the future? 
Can you also update [docs/getting-started-guides/README](docs/getting-started-guides/README.md)?

**Comment by jsafrane** [comment]
Thanks for the review. I'm still trying to get the vagrant setup running reliably, it's very fragile and e2e tests break at random places. I am not sure it's worth the effort. If I get it running, I'll be happy to maintain it.

**Comment by derekwaynecarr** [comment]
Ping me on IRC and I can help diagnose issues.  I have wanted lib virt support so it can do parallel boot up of machines.

**Comment by derekwaynecarr** [comment]
Self assigning as this impacts the virtual box setup

**Comment by derekwaynecarr** [comment]
Closing this PR in favor of 

https://github.com/GoogleCloudPlatform/kubernetes/commit/4753021167e0e4c032e63bd35347d95d462cdb81

**Pull request closed without merging**